File: server/Sprig.Models/Response.cs
namespace Sprig.Models;

using MessagePack;

[MessagePackObject]
public class Response : Message
{
    public Response()
    {
        ResponseCode = Code.Unknown;
    }

    public Response(Code code)
    {
        ResponseCode = code;
    }

    public enum Code
    {
        Unknown,
        Ok,
        GenericError,
        InvalidRequest,
    }

    [Key(1)]
    public Code ResponseCode { get; set; }
}

File: server/Messages/Response.cpp
#include "Response.h"

Response::Response()
	: Code(Kind::Unknown), Message(Message::Kind::Response)
{
}

Response::Response(Response::Kind code)
	: Code(code), Message(Message::Kind::Response)
{
}

Response::Response(Response::Kind code, std::string info)
	: Code(code), Info(std::move(info)), Message(Message::Kind::Response)
{
}
